### PR TITLE
 fix: fix GH Actions workflows sending notification on failure

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
Fix the notifications not being send in the case of failure for Next and Release workflows (docker image removal in the build job requires to put notification action in a separate job)
A special condition is needed in order to ensure that the notification job always is run, but only at success or failed state of the previous build job.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)
- [ ] Optional Companion PR for updating the HappyPath tests is approved and ready to be merged

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: .github/workflows/next-build.yml [hunk 1]
```diff
@@ -10,9 +10,10 @@
 name: Build & Publish `next`
 
 on:
+  workflow_dispatch:
   push:
     branches:
-      - main
+      - main
```
Comment (benoitf) [inline .github/workflows/next-build.yml:16]: Empty Space ?

**Review by benoitf — APPROVED** [review]

File: .github/workflows/next-build.yml [hunk 2]
```diff
@@ -47,14 +48,20 @@ jobs:
...
-        echo "{\"text\":\":no_entry_sign: Next Che Theia build has failed: https://github.com/eclipse-che/che-theia/actions/workflows/next-build.yml\"}" > mattermost.json
-    - name: Send MM message
-      if: ${{ failure() }}
-      uses: mattermost/action-mattermost-notify@1.1.0
-      env:
-        MATTERMOST_WEBHOOK_URL: ${{ secrets.MATTERMOST_WEBHOOK_URL }}
-        MATTERMOST_CHANNEL: eclipse-che-ci
-        MATTERMOST_USERNAME: che-bot
+
+notify:
```
Comment (azatsarynnyy) [inline .github/workflows/next-build.yml:52]: Please, check the indentation. It should be aligned with the `build` job.

File: .github/workflows/next-build.yml [hunk 3]
```diff
@@ -10,9 +10,10 @@
 name: Build & Publish `next`
 
 on:
+  workflow_dispatch:
```
Comment (azatsarynnyy) [inline .github/workflows/next-build.yml:13]: Is there a need to run the workflow manually?

Comment (mkuznyetsov) [reply .github/workflows/next-build.yml:13]: it may be useful for restart the job if necessary (when it previously failed, or for debug purposes)

Comment (azatsarynnyy) [reply .github/workflows/next-build.yml:13]: Agree.

**Pull request merged**